src/cbc_sdk/connection.py

```diff
@@ -488,6 +488,52 @@ def get_raw_data(self, uri, query_parameters=None, default=None, **kwargs):
         else:
             raise ServerError(error_code=result.status_code, message="Unknown error: {0}".format(result.content))
 
+    def get_stream_data(self, uri, stream_output, query_parameters=None, **kwargs):
```

Comment:
This seems like a challenge to maintain in the future as we continue to make more unique functions. Can we determine a better more generic way to combine post_and_get_stream and get_stream_data into a single function? The same goes for iter_lines

Reply:
I really don't think we're going to have to support any HTTP verbs other than GET and POST for stream output like this.

Reply:
This sets a precedence to create a function for each unique verb + mechanism in the future. I'd rather see us follow a model more similar to `api_json_request`. This would help reduce the amount of functions from 4 to 2 request_iter_lines and request_stream. This concern stems from making too many functions which will lead to more confusion over time around what to use. This is always a balance between making a single function do too much and making too many functions however in this case it's the same end result with a slightly different request structure.

Reply:
Well, I call the functions `api_request_stream` and `api_request_iterate`, but that's what's been done.

The function mocks are done in a clever fashion.  Instead of passing 'GET_STREAM' or 'GET_LINES` to `mock_request`, you pass in `STREAM:GET` or `ITERATE:GET`. Same thing with POST requests.  It required minimal modification to mocking the test code.

(And no, we can't combine those further.  One of them takes a stream parameter to write output, the other is an iterator that has to be called with `yield from`. That makes them about as similar as cheese and chalk.)

Reply:
This makes sense and I agree it would not be feasible to combine the functions further